src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -0,0 +1,186 @@
+                             const Commands::AddThermostatSuggestion::DecodableType & commandData)
+{
+
+    // If time is not synced, return INVALID_IN_STATE in the AddThermostatSuggestionResponse.
+    uint32_t currentMatterEpochTimestampInSeconds = 0;
+    if (System::Clock::GetClock_MatterEpochS(currentMatterEpochTimestampInSeconds) != CHIP_NO_ERROR)
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidInState);
+        return true;
+    }
+
+    ChipLogError(Zcl, "current timestamp is %u", currentMatterEpochTimestampInSeconds);
+
+    EndpointId endpoint = commandPath.mEndpointId;
+    auto delegate       = GetDelegate(endpoint);
+    if (delegate == nullptr)
+    {
+        ChipLogError(Zcl, "Delegate is null");
+        commandObj->AddStatus(commandPath, Status::InvalidInState);
+        return true;
+    }
+
+    delegate->SetEndpointId(endpoint);
+
+    // If the preset hande doesn't exist in the Presets attribute, return NOT_FOUND.
+    if (!IsPresetHandlePresentInPresets(delegate, commandData.presetHandle))
+    {
+        commandObj->AddStatus(commandPath, Status::NotFound);
+        return true;
+    }
+
+    // If the thermostat suggestions list is full, return RESOURCE_EXHAUSTED.
+    if (delegate->GetNumberOfThermostatSuggestions() >= delegate->GetMaxThermostatSuggestions())
+    {
+        commandObj->AddStatus(commandPath, Status::ResourceExhausted);
+        return true;
+    }
+
+    // If the effective time in UTC is greater than current time in UTC plus 24 hours, return INVALID_COMMAND.
+    const uint32_t kSecondsInDay = 24 * 60 * 60;
+    if (!commandData.effectiveTime.IsNull() &&
+        (commandData.effectiveTime.Value() > currentMatterEpochTimestampInSeconds + kSecondsInDay))
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+    uint8_t uniqueID = delegate->GetUniqueID();
+
+    Structs::ThermostatSuggestionStruct::Type thermostatSuggestion;
+    thermostatSuggestion.uniqueID     = uniqueID;
+    thermostatSuggestion.presetHandle = commandData.presetHandle;
+
+    uint32_t effectiveTime             = commandData.effectiveTime.ValueOr(currentMatterEpochTimestampInSeconds);
+    thermostatSuggestion.effectiveTime = effectiveTime;
+
+    const uint32_t kSecondsInMinute     = 60;
+    thermostatSuggestion.expirationTime = effectiveTime + (commandData.expirationInMinutes * kSecondsInMinute);
+
+    CHIP_ERROR err = delegate->AppendToThermostatSuggestionsList(thermostatSuggestion);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to AppendToThermostatSuggestionsList with error: %" CHIP_ERROR_FORMAT, err.Format());
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+    MatterReportingAttributeChangeCallback(endpoint, Thermostat::Id, ThermostatSuggestions::Id);
+
+    err = delegate->ReEvaluateCurrentSuggestion();
```

Comment:
Since this doesn't affect the response, it should be done afterwards? May be we only need to set a timer to evaluate later than evaluating it right here.

Reply:
i will move the response before this.